src/simulator/cube.py

```diff
@@ -11,29 +10,42 @@ def __init__(self, length: float, height: float, width: float):
         self.height = height
         self.width = width
 
-    def bounding_box(self) -> Box:
+    def bounding_box(self, position: Position) -> Box:
         """Returns smallest enclosing upright Box
-        """
-
-        # TODO: The following calculation is only for an upright cube
-        # dimension calculations
-        length = self.shape.length
-        height = self.shape.height
-        width = self.shape.width
-
-        # position calculations
-        x = self.position.x
-        y = self.position.y
-        z = self.position.z
-        phi = self.position.phi
-        theta = self.position.theta
 
-        # assign return Arguments
-        bb_cube = Cube(length, height, width)
-        bb_position = Position(x, y, z, phi, theta)
-
-        # return Box(1, bb_cube, bb_position)
-        pass
+        Calculate the smallest bounding box. The function
+        moves the shape to the coordinate origin,calculates
+        the corner points of its dimensions and rotates it.
+        The box is surrounded by the farthest rotated points.
+        Then it returns the shape in the last position. The bounding box
+        has the exact same position as the shape.
+        """
+        l = self.length
+        w = self.width
+        h = self.height
+        # Because the cube has symmetry, we only need the top side points of the disk.
+        # We calculate the distance of the farthest points and then get the dimensions
+        # of the bounding box. Note, since we used the top points, because of symmetry,
+        # we double the dimensions of the bounding box so that the box surrounds the bottom
+        # points of the disk.
+        points = [[l / 2, w / 2, h / 2], [-l / 2, w / 2, h / 2],
+                  [l / 2, -w / 2, h / 2], [-l / 2, -w / 2, h / 2]]
+        new_x = list()
+        new_y = list()
+        new_z = list()
+        for point in points:
+            new_point = Shape.rotation(point[0], point[1], point[2], position.phi, position.theta)
+            new_x.append(new_point[0])
+            new_y.append(new_point[1])
+            new_z.append(new_point[2])
+        length = 2 * max([abs(x) for x in new_x])
```

Comment:
move the comment regarding multiplication by 2 and taking max here. it's better to explain what you do exactly where you do it. Not a few lines earlier.